src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobMetadataStore.cs

```diff
@@ -127,17 +126,79 @@ public Task<DicomDataset> GetInstanceMetadataAsync(VersionedInstanceIdentifier v
 
         return ExecuteAsync(async t =>
         {
-            await using (Stream stream = _recyclableMemoryStreamManager.GetStream(GetInstanceMetadataStreamTagName))
-            {
-                await blobClient.DownloadToAsync(stream, cancellationToken);
+            BlobDownloadResult result = await blobClient.DownloadContentAsync(cancellationToken);
+            return result.Content.ToObjectFromJson<DicomDataset>(_jsonSerializerOptions);
+        }, cancellationToken);
+    }
+
+    /// <inheritdoc />
+    public async Task DeleteInstanceFramesRangeAsync(VersionedInstanceIdentifier versionedInstanceIdentifier, CancellationToken cancellationToken)
+    {
+        EnsureArg.IsNotNull(versionedInstanceIdentifier, nameof(versionedInstanceIdentifier));
+        BlockBlobClient blobClient = GetInstanceFramesRange(versionedInstanceIdentifier);
+
+        await ExecuteAsync(t => blobClient.DeleteIfExistsAsync(DeleteSnapshotsOption.IncludeSnapshots, conditions: null, t), cancellationToken);
+    }
+
+    /// <inheritdoc />
+    public async Task StoreInstanceFramesRangeAsync(
+            VersionedInstanceIdentifier versionedInstanceIdentifier,
+            IReadOnlyDictionary<int, FrameRange> framesRange,
+            CancellationToken cancellationToken)
+    {
+        EnsureArg.IsNotNull(versionedInstanceIdentifier, nameof(versionedInstanceIdentifier));
+        EnsureArg.IsNotNull(framesRange, nameof(framesRange));
 
+        BlockBlobClient blobClient = GetInstanceFramesRange(versionedInstanceIdentifier);
+
+        try
+        {
+            await using (Stream stream = _recyclableMemoryStreamManager.GetStream(StoreInstanceFramesRangeTagName))
```

Comment:
nit: can we use using declarations for async? Like `await using Stream stream = _recyclableMemoryStreamManager.GetStream(StoreInstanceFramesRangeTagName)`?